tools/gitpod/Dockerfile

```diff
@@ -1,74 +1,101 @@
-USER gitpod
+
+# hadolint ignore=DL3008
+RUN apt-get update && \
+    apt-get install -yq --no-install-recommends \
+    ca-certificates \
+    dirmngr \
+    dvisvgm \
+    gnupg \
+    gpg-agent \
+    texlive-latex-extra \
+    vim && \
+    # this needs to be done after installing dirmngr
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-key C99B11DEB97541F0 && \
+    apt-add-repository https://cli.github.com/packages && \
+    apt-get install -yq --no-install-recommends \
+    gh && \
+    locale-gen en_US.UTF-8 && \
+    apt-get clean && \
+    rm -rf /var/cache/apt/* &&\
+    rm -rf /var/lib/apt/lists/* &&\
+    rm -rf /tmp/*
 
 # Allows this Dockerfile to activate conda environments
 SHELL ["/bin/bash", "--login", "-o", "pipefail", "-c"]
 
-# Install mambaforge3
+
+# -----------------------------------------------------------------------------
+# ---- Installing mamba  ----
 RUN wget -q -O mambaforge3.sh \
-    https://github.com/conda-forge/miniforge/releases/download/$MAMBAFORGE_VERSION/Mambaforge-$MAMBAFORGE_VERSION-Linux-x86_64.sh && \
-    bash mambaforge3.sh -p $CONDA_DIR -b && \
+    "https://github.com/conda-forge/miniforge/releases/download/$MAMBAFORGE_VERSION/Mambaforge-$MAMBAFORGE_VERSION-Linux-x86_64.sh" && \
+    bash mambaforge3.sh -p ${CONDA_DIR} -b && \
     rm mambaforge3.sh
 
-# makes conda activate command for this Dockerfile
-RUN echo ". $CONDA_DIR/etc/profile.d/conda.sh" >> ~/.profile
-# enables conda for interactive sessions
-RUN conda init bash
 
-# Install numpy dev dependencies
-COPY --from=clone --chown=gitpod /tmp/numpy /workspace/numpy
-RUN mamba env create -f /workspace/numpy/environment.yml -n $CONDA_ENV && \
-    conda activate $CONDA_ENV && \
-    mamba install ccache -y && \
-    conda clean --all -f -y
+# -----------------------------------------------------------------------------
+# ---- Copy needed files ----
+# basic workspace configurations
+COPY ./tools/gitpod/workspace_config /usr/local/bin/workspace_config
 
-# Set up ccache for compilers for this Dockerfile and interactino sessions
-# Using `conda env config vars set` does not work with Docker
-# REF: https://github.com/conda-forge/compilers-feedstock/issues/31
-RUN echo "conda activate $CONDA_ENV" >> ~/.startuprc && \
-    echo "export CC=\"ccache \$CC\"" >> ~/.startuprc && \
-    echo "export CXX=\"ccache \$CXX\"" >> ~/.startuprc && \
-    echo "export F77=\"ccache \$F77\"" >> ~/.startuprc && \
-    echo "export F90=\"ccache \$F90\"" >> ~/.startuprc && \
-    echo "export GFORTRAN=\"ccache \$GFORTRAN\"" >> ~/.startuprc && \
-    echo "export FC=\"ccache \$FC\"" >> ~/.startuprc && \
-    echo "source ~/.startuprc" >> ~/.profile && \
-    echo "source ~/.startuprc" >> ~/.bashrc
+RUN chmod a+rx /usr/local/bin/workspace_config && \
+    workspace_config
 
-# Build numpy to populate the cache used by ccache
-RUN python /workspace/numpy/setup.py build_ext -i && \
-    ccache -s
+# Copy conda environment file
```

Comment:
Can you add a comment on why this is needed? I don't immediately see a reason to not do `mamba env create -f environment.yml` without the copy.

Reply:
`COPY` copies files or directories from your context directory (in this case the root `scipy`) and adds them to the filesystem of the container 
The `environment.yaml` has to exist inside the container else when doing `mamba env create -f environment.yml` it will return `file environment.yaml does not exist`

Reply:
Added this as a note to the Dockerfile